src/Jellyfin.MediaEncoding.Hls/ScheduledTasks/KeyframeExtractionScheduledTask.cs

```diff
@@ -74,19 +73,16 @@ public Task ExecuteAsync(IProgress<double> progress, CancellationToken cancellat
             query.StartIndex = startIndex;
 
             var videos = _libraryManager.GetItemList(query);
-            var currentPageCount = videos.Count;
-            // TODO parallelize with Parallel.ForEach?
-            for (var i = 0; i < currentPageCount; i++)
+            foreach (var video in videos)
             {
-                var video = videos[i];
                 // Only local files supported
-                if (video.IsFileProtocol && File.Exists(video.Path))
+                var path = video.Path;
+                if (!string.IsNullOrEmpty(path) && File.Exists(path))
```

Comment:
File.Exists already checks for null and empty
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/IO/File.cs,3360368484a9f131
```suggestion
                if (File.Exists(path))
```